### PR TITLE
WIP: convert to use ES modules, setup unit tests with karma

### PR DESCRIPTION
**Pull request by web-padawan**
Connected to vaadin/components-team-tasks#424

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-element-skeleton/182)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 sudo: false
 dist: trusty
```
Comment (limonte) [inline .travis.yml]: `xenial` is available now: https://blog.travis-ci.com/2018-11-08-xenial-release

File: .travis.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
 language: node_js
-node_js: 8.11
+node_js: node
 
 cache:
```
Comment (limonte) [inline .travis.yml]: this might cause issues, see https://github.com/kenwheeler/cash/pull/233 for details

**Comment by limonte** [comment]
Any reason not to use `yarn.lock`? There more than 2 thousands of dev-deps and the recent accident with `event-stream` package demonstrated why lock files are useful, read more: https://schneid.io/blog/event-stream-vulnerability-explained/

btw `npm install` results with

> found 5 vulnerabilities (1 low, 4 high)

It's not a wise practice to just ignore them.

File: .travis.yml (hunk 2 above)
Comment (web-padawan) [reply .travis.yml]: Removed.

File: .travis.yml (hunk 1 above)
Comment (web-padawan) [reply .travis.yml]: Updated.

File: .travis.yml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 sudo: false
 dist: trusty
 language: node_js
-node_js: 8.11
+node_js: node
```
[the remark this answers is not in the document]
Comment (web-padawan) [reply .travis.yml]: Done.

**Comment by web-padawan** [comment]
Review feedback addressed. For now moving back to "Design & discussion" as agreed.

The topics left out of the initial scope are:
- [ ] visual tests
- [ ] test-fixture replacement
- [ ] stylelint (possible if we use `lit-css and `StyledLitElement`)

**Comment by moderndeveloperllc** [comment]
If you are going to use eslint and lit elements, might want to check out adding [eslint-plugin-lit](https://www.npmjs.com/package/eslint-plugin-lit) to the babel parser plugins list. has a decent collection of rules to help with LitElement linting.

**Comment by web-padawan** [comment]
@moderndeveloperllc yes, Vaadin platform 15 will drop IE11 support and in practice that means that next major versions for all our components will not be tested against it anymore.

**Comment by web-padawan** [comment]
Closing in favor of #207 which contains a rebased and squashed version of this PR.

**Pull request closed without merging**